app/server.js

```diff
@@ -0,0 +1,204 @@
+'use strict'
+
+let util = require('util');
+let express = require('express');
+let passport = require('passport');
+let GitHubStrategy = require('passport-github2').Strategy;
+let LocalStrategy = require('passport-local').Strategy;
+let bodyParser = require('body-parser');
+let partials = require('express-partials');
+let methodOverride = require('method-override');
+let session = require('express-session');
+let request = require('request');
+let http = require('http');
+let coForEach = require('co-foreach');
+let Q = require('q');
+var co = require('co');
+let RequestClient = require('./requestService');
+let requestClient = new RequestClient();
+let User = require('./user.js');
+let DBService = require('./dbService');
+let dbService = new DBService();
+
+const ensureAuthenticated = require('./middlewares/ensureAuthenticated');
+
+//postgresql://[user[:password]@][netloc][:port][/dbname][?param1=value1&...]
+// let connectionString = process.env.DATABASE_URL || 'postgresql://postgres:postgres@localhost/githubDB';
+
+let GITHUB_CLIENT_ID = '5301c2ab0614cc72f15c';
+let GITHUB_CLIENT_SECRET = 'be52db4573cf12873a57117e59848307e0f0a37d';
+
+passport.serializeUser((user, done) => done(null, user));
+
+passport.deserializeUser((obj, done) => done(null, obj));
+
+passport.use(new GitHubStrategy({
+  clientID: GITHUB_CLIENT_ID,
+  clientSecret: GITHUB_CLIENT_SECRET,
+  callbackURL: "http://127.0.0.1:3000/auth/github/callback"
+},
+  (accessToken, refreshToken, profile, done) => {
+    profile._json.token = accessToken; 
+
+    process.nextTick(() => {
+      return done(null, profile);
+    });
+  }
+));
+
+passport.use(new LocalStrategy( (username, password, done)=> {
+  printMessage(username, password);
+
+
+  done();
+  // query database to get user details
+  
+}))
+
+
+// configure Express
+var app = express();
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+app.use(partials());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+app.use(methodOverride());
+
+app.use(session({ secret: 'adam alinoe', resave: false, saveUninitialized: false }));
+
+app.use(passport.initialize());
+app.use(passport.session());
+app.use(express.static(__dirname + '/public'));
+
+app.get('/', (req, res) => res.render('index', { user: req.user }));
+
+app.get('/login', (req, res) => {
+  res.render('login', { user: req.user });
+});
+
+app.post('/login-user', (req, res) => {
+  passport.authenticate('local', {successRedirect: '/account', failureRedirect: '/login', failureFlash: true })
+
+})
+
+app.get('/auth/github', passport.authenticate('github', { scope: ['user:email'] }), (req, res) => { /*tutaj nigdy nie bedziemy, idz do callback*/ });
+
+app.get('/auth/github/callback', passport.authenticate('github', { failureRedirect: '/login' }), (req, res) => {
```

Comment:
co robi ten endpoint? To zdecydowanie nie jest REST. 

Zwraca jakiś callback z githuba autoryzacji?